document/internal/serde/field.go

```diff
@@ -0,0 +1,292 @@
+package serde
```

Comment:
The only reasonable way I can think of would require us to just make this a publicly accessible Go package in smithy-go and we would have to maintain the public interfaces and forward compatability. It's certainly something we could do, and it would likely probably best done as a stand-alone module. Would we want to do that now or later as a separate task?

Reply:
Makes sense. That sounds like something we could add later as some kind of type walker helper if we decided its valuable in a maintainable way. We probably don't want to do that here.